python/paddle/distributed/utils.py

```diff
@@ -0,0 +1,450 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import paddle.fluid as fluid
+import logging
+import socket
+import time
+import os
+import signal
+import copy
+import sys
+import subprocess
+from contextlib import closing
+import socket
+
+logger = logging.getLogger("root")
+logger.propagate = False
+
+
+class Hdfs(object):
+    def __init__(self):
+        self.hdfs_ugi = None
+        self.hdfs_name = None
+        self.hdfs_path = None
+
+    def is_valid(self):
+        return self.hdfs_ugi is not None and \
+            self.hdfs_name is not None and \
+            self.hdfs_path is not None
+
+    def __str__(self):
+        return "hdfs_ugi:{} hdfs_name:{} hdfs_path{}".format(
+            self.hdfs_ugi, self.hdfs_name, self.hdfs_path)
+
+    def __eq__(self, n):
+        return self.hdfs_ugi == n.hdfs_ugi and \
+            self.hdfs_name == n.hdfs_name and \
+            self.hdfs_path == n.hdfs_path
+
+    def __ne__(self, n):
+        return not self == n
+
+
+class Cluster(object):
```

Comment:
These objects do not seem to be `utils`. Is it better to aggregate them to a new file?

Reply:
The cloud cluster use them also.

Reply:
In my opinion, `split_string`, `atoi`, etc. are utils, while `Cluster`, `Pod` are something like `Program`,  `Block`, and they are in framework domain.